app/src/processing/app/ui/ChangeDetector.java

```diff
@@ -150,39 +168,111 @@ private synchronized void checkFiles() {
+                sketch.loadNewTab(filenames.get(i), extensions.get(i), true);
+                break;
+              }
             }
-            try {
-              sketch.save();
-            } catch (Exception e) {
-              //if that didn't work, tell them it's un-recoverable
-              Messages.showError("Reload Failed", "The main file for this sketch was deleted\n" +
-                  "and could not be rewritten.", e);
+          }
+          for (SketchCode modifiedCode : modifiedCodes) {
+            if (!mergeConflicts.contains(modifiedCode)) {
+              sketch.loadNewTab(modifiedCode.getFileName(),
+                  modifiedCode.getExtension(), false);
+              break;
             }
           }
 
+          // Destructive actions, so prompt.
+          if (ask) {
+            sketch.updateSketchCodes();
+
+            showReloadPrompt(mergeConflicts, removedCodesFinal,
+              scReload -> {
+                try {
+                  File file = scReload.getFile();
+                  File autosave = File.createTempFile(scReload.getPrettyName(),
+                    ".autosave", file.getParentFile());
+                  scReload.copyTo(autosave);
+                } catch (IOException e) {
+                  Messages.showWarning("Could not autosave modified tab",
+                      "Your changes to " + scReload.getPrettyName() +
+                      " have not been saved, so we won't load the new version.", e);
+                  scReload.setModified(true); // So we'll have another go at saving
+                                              // it later,
+                  ignoredModifications.add(scReload); // but not create a loop.
+                  return;
+                }
+                sketch.loadNewTab(scReload.getFileName(), scReload.getExtension(), false);
+              },
+              scKeep -> {
+                try {
+                  File file = scKeep.getFile();
+                  File autosave = File.createTempFile(scKeep.getPrettyName(),
+                    ".autosave", file.getParentFile());
+                  // It is platform-dependent whether File.renameTo would let
+                  // you overwrite the dummy file autosave, which is used to
+                  // make sure all autosave files are unique.
+                  Files.move(file.toPath(), autosave.toPath(),
+                      java.nio.file.StandardCopyOption.REPLACE_EXISTING);
+                } catch (IOException e) {
+                  Messages.showWarning("Move failed",
+                      "Could not move the external editor's"
+                    + " version of " + scKeep.getPrettyName() + " to a safe"
+                    + " location; make a copy of it before saving the sketch"
+                    + " if you need it.", e);
+                  ignoredModifications.add(scKeep); // No infinite loops.
+                  return;
+                }
+                try {
+                  scKeep.save();
```

Comment:
@benfry Do we want to save the tab right here if the user decides not to reload external changes? Other text editors don't save. Maybe updating the timestamp of the SketchCode to the timestamp of the file (so it won't pop up again until the file is changed externally again) and marking the SketchCode as modified is more what users are used to?